compose/config/config.py

```diff
@@ -668,7 +734,13 @@ def is_url(build_path):
 
 def validate_paths(service_dict):
     if 'build' in service_dict:
-        build_path = service_dict['build']
+        build = service_dict.get('build', {})
+
+        if isinstance(build, six.string_types):
+            build_path = build
+        elif isinstance(build, dict) and 'context' in build:
+            build_path = build['context']
```

Comment:
I would expect that `build` would be normalized to always be a `dict()` by the time it gets here, do we really need the string case still?

If we do need it, could we have `process_service()` always return a dict for `build` so we don't have to worry about the variation after that point.